### PR TITLE
Exit with code 1 when deploy failed (fixes #324)

### LINKED ISSUE #324: Codeship does not fail when `mup deploy` fails
In Codeship I have configured deployment via a custom script with the content `mup deploy`. I use mup 0.9.6.

Here is the log of a failing deploy where Codeship still reports success: https://gist.github.com/Sanjo/6c702cd17db1ab5422e7.

How can I make Codeship fail when this happens?

### PR DESCRIPTION
**Pull request by ghost**
I've tested it manually.

### CODE REVIEW

File: lib/actions.js [hunk 1]
```diff
@@ -126,6 +126,8 @@ Actions.prototype.deploy = function() {
 
   function afterCompleted(summeryMap) {
```
Comment (arunoda) [inline lib/actions.js]: This `afterCompleted` method will be called multiple times.
So, we need to wait until others getting completed before doing `process.exit()`.

Comment (ghost) [inline lib/actions.js]: Can I add [async](https://www.npmjs.com/package/async) as dependency to handle this?

Comment (arunoda) [inline lib/actions.js]: Okay.

**Comment by ghost** [comment]
Need to test the updated pull request later. I only deploy to one server though.

**Comment by arunoda** [comment]
Okay. I'll do it.

**Comment by ghost** [comment]
:+1:

**Comment by ghost** [comment]
I've done another mistake. I will fix it and tell you when it works.

**Comment by arunoda** [comment]
Just tell it :)

**Comment by ghost** [comment]
Ok, now.

File: lib/actions.js [hunk 2]
```diff
@@ -111,20 +112,37 @@ Actions.prototype.deploy = function() {
...
+          });
+        },
+        exit
+      )
     }
   });
 
-  function afterCompleted(summeryMap) {
+  function afterCompleted(summaryMap) {
     rimraf.sync(buildLocation);
```
Comment (arunoda) [inline lib/actions.js:146]: You need to remove this and push it to the end.
I've done that. No need to do a commit.

**Comment by arunoda** [comment]
As a by product of this, now we've rolling deployment support when we've multiple servers. 
One server at a time. :D

**Comment by arunoda** [comment]
Published: 0.9.7

**Pull request merged**